### PR TITLE
[type:refactor] refactor shenyu-client-grpc

### PR DESCRIPTION
**Pull request by totalo**
For #3932 

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by dragon-zhang** [comment]
please check CI

**Comment by dragon-zhang** [comment]
企业微信截图_1800ce9c-275e-419b-abc6-6292c0a57ecf

**Comment by totalo** [comment]
> 企业微信截图_1800ce9c-275e-419b-abc6-6292c0a57ecf

Thanks, I noticed, currently only testing as a draft.

File: shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientEventListener.java [hunk 1]
```diff
@@ -34,165 +34,136 @@
...
-
+    
     @Override
-    public void onApplicationEvent(@NonNull final ContextRefreshedEvent contextRefreshedEvent) {
-        Map<String, BindableService> beans = contextRefreshedEvent.getApplicationContext().getBeansOfType(BindableService.class);
-        for (Map.Entry<String, BindableService> entry : beans.entrySet()) {
+    protected Map<String, BindableService> getBeans(final ApplicationContext context) {
+        Map<String, BindableService> result = context.getBeansOfType(BindableService.class);
+        for (Entry<String, BindableService> entry : result.entrySet()) {
             exportJsonGenericService(entry.getValue());
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientEventListener.java:75]: `exportJsonGenericService(entry.getValue());` should be put to `#handle`.

Comment (totalo) [reply shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientEventListener.java]: done.

**Comment by dragon-zhang** [comment]
image
I ran the version you modified and the master branch version. There is a problem when handle with `/**`, please fix it.

**Comment by totalo** [comment]
hi，@dragon-zhang, I have fixed the issue with `helloService/**`. And for `access$000` it is because there is a place in the inner class to call the method of the outer class through the this reference of the outer class. The `access$000` method is a method automatically generated by the compiler because the `onError` method of the anonymous inner class in the `helloEveryOne` method accesses the `Log` object of the outer class. And only when the annotation is on the class
Since there is no way to avoid business implementation, I think our solution may be as follows for correctness:
1. Only configure the generation of non-static methods
2. Generated only for public methods
I prefer 2, what you think?

**Comment by dragon-zhang** [comment]
> hi，@dragon-zhang, I have fixed the issue with `helloService/**`. And for `access$000` it is because there is a place in the inner class to call the method of the outer class through the this reference of the outer class. The `access$000` method is a method automatically generated by the compiler because the `onError` method of the anonymous inner class in the `helloEveryOne` method accesses the `Log` object of the outer class. And only when the annotation is on the class Since there is no way to avoid business implementation, I think our solution may be as follows for correctness:
> 
> 1. Only configure the generation of non-static methods
> 2. Generated only for public methods
> 
> I prefer 2, what you think?

agree with you, just go ahead.

**Comment by totalo** [comment]
> ahead

Got it, You can review it, the current one is handled according to public.

**Comment by totalo** [comment]
> > hi，@dragon-zhang, I have fixed the issue with `helloService/**`. And for `access$000` it is because there is a place in the inner class to call the method of the outer class through the this reference of the outer class. The `access$000` method is a method automatically generated by the compiler because the `onError` method of the anonymous inner class in the `helloEveryOne` method accesses the `Log` object of the outer class. And only when the annotation is on the class Since there is no way to avoid business implementation, I think our solution may be as follows for correctness:
> > 
> > 1. Only configure the generation of non-static methods
> > 2. Generated only for public methods
> > 
> > I prefer 2, what you think?
> 
> agree with you, just go ahead.

Got it, You can review it, the current one is handled according to public.

**Review by dragon-zhang — APPROVED** [review]

**Comment by dragon-zhang** [comment]
Thanks for your contribution !

**Pull request merged**